xray/plot/plot.py

```diff
@@ -424,27 +438,63 @@ def newplotfunc(darray, ax=None, xincrease=None, yincrease=None,
         if ax is None:
             ax = plt.gca()
 
-        # Handle the dimensions
-        try:
-            ylab, xlab = darray.dims
-        except ValueError:
-            raise ValueError('{name} plots are for 2 dimensional DataArrays. '
+        dims = list(darray.dims)
+
+        if len(dims) != 2:
+            raise ValueError('{type} plots are for 2 dimensional DataArrays. '
                              'Passed DataArray has {ndim} dimensions'
-                             .format(name=plotfunc.__name__, ndim=len(darray.dims)))
+                             .format(type=plotfunc.__name__, ndim=len(dims)))
+
+        if x and x not in darray:
+            raise KeyError('{0} is not a dimension of this DataArray. Use '
```

Comment:
So then you could do things like this:

```
In [1]: a = xray.DataArray(np.random.randn(4, 5))

In [2]: a.coords['z'] = 100

In [4]: a.plot.contourf(x='dim_0', y='z')
```

This doesn't really make sense to me if 'z' is not a dimension for a.